sql-plugin/src/main/scala/org/apache/spark/sql/rapids/mathExpressions.scala

```diff
@@ -419,6 +419,68 @@ case class GpuCot(child: Expression) extends GpuUnaryMathExpression("COT") {
   }
 }
 
+object GpuHypot extends Arm {
+  def chooseXandY(lhs: ColumnVector, rhs: ColumnVector): Seq[ColumnVector] = {
+    withResource(lhs.abs) { lhsAbs =>
+      withResource(rhs.abs) { rhsAbs => 
+        withResource(lhsAbs.greaterThan(rhsAbs)) { lhsGreater =>
+          closeOnExcept(lhsGreater.ifElse(lhsAbs, rhsAbs)) { x => 
+            Seq(x, lhsGreater.ifElse(rhsAbs, lhsAbs))
+          }
+        }
+      }
+    }
+  }
+
+  def either(value: Scalar, x: ColumnVector, y: ColumnVector): ColumnVector = {
+    withResource(x.equalTo(value)) { xIsVal =>
+      withResource(y.equalTo(value)) { yIsVal =>
+        xIsVal.or(yIsVal)
+      }
+    }
+  }
+
+  def both(value: Scalar, x: ColumnVector, y: ColumnVector): ColumnVector = {
+    withResource(x.equalTo(value)) { xIsVal =>
+      withResource(y.equalTo(value)) { yIsVal =>
+        xIsVal.and(yIsVal)
+      }
+    }
+  }
+
+  def eitherNan(x: ColumnVector, 
+                y: ColumnVector): ColumnVector = {
+    withResource(x.isNan) { xIsNan =>
+      withResource(y.isNan) { yIsNan =>
+        xIsNan.or(yIsNan)
+      }
+    }
+  }
+
+  def eitherNull(x: ColumnVector,
+                 y: ColumnVector): ColumnVector = {
+    withResource(x.isNull) { xIsNull =>
+      withResource(y.isNull) { yIsNull =>
+        xIsNull.or(yIsNull)
+      }
+    }
+  }
+
+  def computeHypot(x: ColumnVector, y: ColumnVector): ColumnVector = {
+    withResource(y.div(x)) { yOverX =>
```

Comment:
Nit: there are a lot of intermediate values that live longer than they need to.  Could we do it a bit more like.

```
    val yOverXSquared = withResource(y.div(x)) { yOverX =>
        yOverX.mul(yOverX)
    }
    val onePlusTerm = withResource(yOverXSquared) { yOverXSquared =>
      withResource(Scalar.fromDouble(1)) { one =>
        one.add(yOverXSquared)
      }
    }
    val onePlusTermSqrt = withResource(onePlusTerm) { onePlusTerm =>
      onePlusTerm.sqrt
    }
    withResource(onePlusTermSqrt) { onePlusTermSqrt =>
      x.mul(onePlusTermSqrt)
    }
```

Reply:
Cleaned up in latest commit